core/src/abstractions.rs

```diff
@@ -36,8 +36,8 @@ pub(crate) struct MeteredPermitDealer<SK: SlotKind> {
     /// is at this number, no more permits will be requested from the supplier until one is freed.
     /// This avoids requesting slots when we are at the workflow cache size limit. If and when
     /// we add user-defined cache sizing, that logic will need to live with the supplier and
-    /// there will need to be some associated refactoring.
-    max_permits: Option<usize>,
+    /// there will need to be some associated refactoring. // TODO: sounds like we'll need to do this
+    max_permits: Option<Arc<AtomicUsize>>,
```

Comment:
Potentially, yes. Let's leave proper implementations of the commands for later though, since this PR is already quite large, and they won't be working server side for a while anyway.